install/kubernetes/helm/istio/charts/security/templates/deployment.yaml

```diff
@@ -60,12 +60,28 @@ spec:
           {{- if .Values.global.trustDomain }}
             - --trust-domain={{ .Values.global.trustDomain }}
           {{- end }}
+          {{- if .Values.healthChecking }}
+            - --liveness-probe-path=/tmp/ca.liveness # path to the liveness health check status file
+            - --liveness-probe-interval=60s # interval for health check file update
+            - --probe-check-interval=15s    # interval for health status check
+          {{- end }}
           livenessProbe:
+          {{- if .Values.healthChecking }}
+            exec:
+              command:
+              - /usr/local/bin/istio_ca
+              - probe
+              - --probe-path=/tmp/ca.liveness # path to the liveness health check status file
+              - --interval=125s # the maximum time gap allowed between the file mtime and the current sys clock
```

Comment:
One question, will the `interval` here will conflict with the default `periodSeconds` of liveness check?

Reply:
Actually we explicitly configured the `periodSeconds` to 60s to match the interval (125s) and lower the overhead. Please see https://istio.io/docs/tasks/security/health-check/ for the details.

Reply:
got it.

Reply:
The helm flag name is not quite informative. Follow the template logic, it's

when this flag is turned on we use `cmd_exec_correct_liveness_health_check`
otherwise, we use `dummy_http_version_endpoint`.

Also `service health checking` has its own meaning in k8s service discovery (AFAIK), here we're doing liveness health check on a deployment. So might be better to change to

```
  livenessHealthCheck:
{{ if  .Values.livenessHealthCheckEnabled }}
     # the cmd exec
{{ else }}
  # with todo on removing this after 1.2, doing it in master without CP.
 httpGet:
   /version
{{ end }}
```

Reply:
I'll use `citadelServiceHealthCheckEnabled`. I think this avoids most confusions.

Reply:
Changed to `citadelHealthCheck`. It doesn't cause confusion as we removed the other httpGet health check. Sorry for the back and forth.